k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -7,12 +7,42 @@ use crate::common::{
 };
 use k8s_openapi::api::core::v1::Pod;
 use kube::{api::ObjectList, ResourceExt};
+use openapi::models::{Volume, VolumeStatus};
 use semver::{Version, VersionReq};
 use snafu::ResultExt;
+use std::time::Duration;
 use tracing::{info, warn};
 
 /// Function to check for any volume rebuild in progress across the cluster
 pub(crate) async fn is_rebuilding(rest_client: &RestClientSet) -> Result<bool> {
+    loop {
+        let unhealthy_volumes = list_unhealthy_volumes(rest_client).await?;
+        if unhealthy_volumes.is_empty() {
+            break;
+        }
+
+        for volume in unhealthy_volumes.iter() {
+            match replica_rebuild_count(volume.clone()).await {
+                0 => {
+                    for _i in 0 .. 11 {
+                        // wait for a minute for any rebuild to start
+                        tokio::time::sleep(Duration::from_secs(60_u64)).await;
+                        let count = replica_rebuild_count(volume.clone()).await;
+                        if count > 0 {
```

Comment:
In the last attempt (10 mins passed) if we see there is no rebuild. We return Ok(true). We do no exclude this volume for which we already waited 10 mins? This volume will again be listed as rebuild didnt start and we wait again?